### PR TITLE
Fixed rename of taxons

### LINKED ISSUE #9031: "Rename" function broken (taxonomies)
## Description

There are different ways one can rename taxonomies. By visiting `/admin/taxonomies/<taxon_id>/edit`, one can:
i) edit the Name field
ii) right click on the taxon; select Edit
iii) right click on the taxon; select Rename

This issue is about option iii): it currently takes no effect whatsoever.

## Expected Behavior

Editing a taxon name by right clicking on the taxon and selecting Rename should change its name.

## Actual Behaviour

Right clicking on the taxon and selecting Rename takes no effect.

## Steps to Reproduce

As superadmin:
1. Visit `/admin/taxonomies/<taxon_id>/edit`
2. Right click on the taxon
3. Select Rename and change the name
4. Click Update, or on the "Back to Taxonomies list"
5. Check whether the change took effect -> see that it didn't

## Animated Gif/Screenshot

image

## Workaround

You can still change the taxonomy name by using steps i) and ii) (see Description section, above)

## Severity

bug-s3: a feature is broken but there is a workaround

## Your Environment

* Version used: v4.1.22
* Browser name and version: Firefox 96
* Operating System and version (desktop or mobile): Ubuntu 20.04

## Possible Fix

Maybe related to:
https://github.com/openfoodfoundation/openfoodnetwork/pull/8678

### PR DESCRIPTION
**Pull request by brittneyhoneycutt**
#### What? Why?

Closes #9031

Currently, right click on the taxon and selecting Rename does not work. This is a fix for the renaming of a taxon in a tree, not necessarily the name of the entire tree. 

#### What should we test?
As a superadmin:
- Visit /admin/taxonomies/<taxon_id>/edit
- Right click on the taxon (not the name of the tree)
- Select Rename and change the name
- Click Update, or on the "Back to Taxonomies list"
- Check whether the change took effect -> see that it now updates the name of the taxon

#### Release notes

Changelog Category: Technical changes

Fixed Rename of Taxon Functionality

### CODE REVIEW

**Review by mkllnk — APPROVED** [review]
Thanks!

Do you think you could write a system spec for this? You could add to `spec/system/admin/configuration/taxonomies_spec.rb`.

File: app/assets/javascripts/admin/spree/taxons/taxonomy.js.coffee [hunk 1]
```diff
@@ -40,8 +40,8 @@ handle_rename = (e, data) ->
   node = data.rslt.obj
   name = data.rslt.new_name
 
-  url = Spree.url(base_url).clone()
-  url.pathname = url.pathname + '/' + node.attr("id")
+  url =  $.extend(true, {}, base_url)
```
Comment (jibees) [inline app/assets/javascripts/admin/spree/taxons/taxonomy.js.coffee:43]: I'd vote to avoid jQuery every time this is possible.
Here, I think `$.extend()` could easily be replaced by `Object.assign()`.

```suggestion
  url = Object.assign(base_url, {})
```

File: app/assets/javascripts/admin/spree/taxons/taxonomy.js.coffee [hunk 2]
```diff
@@ -40,8 +40,8 @@ handle_rename = (e, data) ->
...
+  url = url['href'] + '/' + node.attr("id")
```
Comment (jibees) [inline app/assets/javascripts/admin/spree/taxons/taxonomy.js.coffee:44]: Why did you change that? 
Before we only update the `pathname` attribute of that URL object.
Here we entirely reassign a new object to a String (and therefore the copy the line below is not needed).
I think the previous code is functional.

**Review by jibees — CHANGES_REQUESTED** [review]
Thanks @brittneyhoneycutt !

I have two suggestions, and I'm wondering why you didn't change the `handle_delete` and `handle_move` methods as well?

Also, I think that renaming the first taxon of the list is still not functional (point 2 on https://github.com/openfoodfoundation/openfoodnetwork/issues/9031#issuecomment-1098804576)

As @mkllnk said, a spec would have been great too!

File: app/assets/javascripts/admin/spree/taxons/taxonomy.js.coffee (hunk 2 above)
Comment (brittneyhoneycutt) [reply app/assets/javascripts/admin/spree/taxons/taxonomy.js.coffee:44]: Thank you for the suggestion! I will be sure to make that change to the pathname attribute, instead. I've just committed your request changes for the first suggestions. My team was unable to determine how to rename the first taxon in the list as the input form takes precedent. Handle_delete and handle_move will have similar solutions. I am currently still at university; thus, I have very limited resources. In a few weeks, I should have more availability to fix these issues as well, if they are still open 👍

Comment (jibees) [reply app/assets/javascripts/admin/spree/taxons/taxonomy.js.coffee]: Ok, I've opened an issue then: https://github.com/openfoodfoundation/openfoodnetwork/issues/9113

File: app/assets/javascripts/admin/spree/taxons/taxonomy.js.coffee [hunk 3]
```diff
@@ -41,12 +41,12 @@ handle_rename = (e, data) ->
...
 
   url = Object.assign(base_url, {})
-  url = url['href'] + '/' + node.attr("id")
+  url.pathname = url.pathname + '/' + node.attr("id")
 
   $.ajax
     type: "POST",
     dataType: "json",
-    url: url.toString(),
+    url: url.pathname.toString(),
```
Comment (jibees) [inline app/assets/javascripts/admin/spree/taxons/taxonomy.js.coffee:49]: I don't think this one is necessary. I'd vote to keep as it, to have minimal modification.

**Review by jibees — CHANGES_REQUESTED** [review]
Hey @brittneyhoneycutt 
Thanks for your contribution !
Two more things: 
 - I've commented around `url: url.pathname.toString(),`
 - I think all those three commits (+ another one?) should be squashed into one, to get a clean/concise/self-explained  git history

**Comment by sigmundpetersen** [comment]
Closing as no update for a while.
Please just give us a ping of you would like to work on it again @brittneyhoneycutt
